core/domain/stats_domain.py

```diff
@@ -1351,3 +1352,115 @@ def validate(self):
             raise utils.ValidationError(
                 'calculation_output is too big to be stored (size: %d): %s' % (
                     sys.getsizeof(output_data), str(output_data)))
+
+
+class LearnerAnswerDetails(object):
+    """Domain object that represents the answer details submitted by the
+    learner.
+    """
+
+    def __init__(
+            self, state_reference, entity_type, interaction_id,
+            learner_answer_info_list, schema_version=(
+                feconf.CURRENT_LEARNER_ANSWERS_DETAILS_SCHEMA_VERSION)):
+        """Constructs a LearnerAnswerDetail domain object.
+
+            Args:
+                state_reference: str. The reference to the state.
+                entity_type: str. The type of entity, eg. exploration,
```

Comment:
You're leaving this too open. Have named constants ENTITY_TYPE_EXPLORATION and ENTITY_TYPE_QUESTION and say that this must be one of [ENTITY_TYPE_EXPLORATION, ENTITY_TYPE_QUESTION].

Reply:
Done